File: Go/Cryptography/get-hmac-message-authentication.go
package main

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/hex"
	"fmt"
)

func main() {
	// Get the hmac message authentication
	fmt.Println(getHMACMessageAuthentication([]byte("Hello, World!"), []byte("password")))
}

// Get the hmac message authentication
func getHMACMessageAuthentication(content []byte, password []byte) string {
	hash := hmac.New(sha256.New, password)
	hash.Write(content)
	return hex.EncodeToString(hash.Sum(nil))
}

File: Dart/Cryptography/get_hmac_message_authentication.dart
// Import the libraries needed to use the HMAC and encoding functions
import "dart:convert";
import "dart:core";
import "package:crypto/crypto.dart";

// Define the main function
void main() {
  // Call the getHMACMessageAuthentication function and print the result
  print(getHMACMessageAuthentication("Hello, World!", "password"));
}

// Define the function that generates the HMAC message authentication code
Digest getHMACMessageAuthentication(
  final String content,
  final String password,
) {
  // Encode the password in UTF-8 format
  final List<int> passwordBytes = utf8.encode(password);
  
  // Create an HMAC object using the SHA-256 algorithm and the password bytes
  final Hmac hmac = Hmac(sha256, passwordBytes);

  // Encode the content string in UTF-8 format
  final List<int> contentBytes = utf8.encode(content);

  // Compute the HMAC for the content string using the password HMAC object
  final Digest digest = hmac.convert(contentBytes);

  // Return the resulting digest as a Digest object
  return digest;
}
